### PR TITLE
refact: moving websockets code to .Shared to prepare for client impl.

### PR DESCRIPTION
**Pull request by valbers**
We intend to implement the client part now. It's useful to share contract-related code between client and server implementations.

### CODE REVIEW

File: src/FSharp.Data.GraphQL.Shared/Serialization/JsonConverters.fs [hunk 1]
```diff
@@ -18,7 +19,7 @@ type ClientMessageConverter () =
     let invalidMsg (explanation : string) = InvalidMessage (4400, explanation) |> Result.Error
 
     let errMsgToStr (struct (docId : int, graphQLErrorMsgs : GQLProblemDetails list)) =
-        String.Join ('\n', graphQLErrorMsgs |> Seq.map (fun err -> err.Message))
+        String.Join ("n", graphQLErrorMsgs |> Seq.map (fun err -> err.Message))
```
Comment (valbers) [inline src/FSharp.Data.GraphQL.Shared/Serialization/JsonConverters.fs]: Apparently there's no `Join` overload for using a character as separator when targeting .netstandard2.0.

File: src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLSubscriptionsManagement.fs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module internal FSharp.Data.GraphQL.Server.AspNetCore.GraphQLSubscriptionsManagement
 
-open FSharp.Data.GraphQL.Server.AspNetCore.WebSockets
+open FSharp.Data.GraphQL.Shared.Websockets
```
Comment (xperiandri) [inline src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLSubscriptionsManagement.fs]:
```suggestion
open FSharp.Data.GraphQL.Shared.WebSockets
```

File: src/FSharp.Data.GraphQL.Shared/Output.fs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+namespace FSharp.Data.GraphQL.Shared
+
+open System.Collections.Generic
+
+type Output = IDictionary<string, obj>
```
Comment (xperiandri) [inline src/FSharp.Data.GraphQL.Shared/Output.fs]: I would put `NameValueLookup` into this file

File: src/FSharp.Data.GraphQL.Shared/Websockets/Exceptions.fs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+namespace FSharp.Data.GraphQL.Shared.Websockets
```
Comment (xperiandri) [inline src/FSharp.Data.GraphQL.Shared/Websockets/Exceptions.fs]:
```suggestion
namespace FSharp.Data.GraphQL.Shared.WebSockets
```

File: src/FSharp.Data.GraphQL.Shared/Websockets/Messages.fs [hunk 5]
```diff
@@ -1,10 +1,9 @@
-namespace FSharp.Data.GraphQL.Server.AspNetCore.WebSockets
+namespace FSharp.Data.GraphQL.Shared.Websockets
```
Comment (xperiandri) [inline src/FSharp.Data.GraphQL.Shared/Websockets/Messages.fs]:
```suggestion
namespace FSharp.Data.GraphQL.Shared.WebSockets
```

File: src/FSharp.Data.GraphQL.Shared/Websockets/Messages.fs
Comment (xperiandri) [inline src/FSharp.Data.GraphQL.Shared/Websockets/Messages.fs:1]: I would put `InvalidWebsocketMessageException` into this file and name it `WebSockets.fs`

File: src/FSharp.Data.GraphQL.Shared/Output.fs (hunk 3 above)
Comment (valbers) [reply src/FSharp.Data.GraphQL.Shared/Output.fs]: Don't you think it's kind of confusing to name the file `Output.fs` and there's just one line of code defining `Output` as an alias for `Dictionary<string, obj>` while the rest of the file, which is not small by the way, is all just the definition of `NameValueLookup` 🙂 ?

File: src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLSubscriptionsManagement.fs (hunk 2 above)
Comment (valbers) [reply src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLSubscriptionsManagement.fs]: Done.

File: src/FSharp.Data.GraphQL.Shared/Websockets/Messages.fs (hunk 5 above)
Comment (valbers) [reply src/FSharp.Data.GraphQL.Shared/Websockets/Messages.fs]: Done.

File: src/FSharp.Data.GraphQL.Shared/Websockets/Messages.fs
Comment (valbers) [reply src/FSharp.Data.GraphQL.Shared/Websockets/Messages.fs:1]: Done.

File: src/FSharp.Data.GraphQL.Shared/Websockets/Exceptions.fs (hunk 4 above)
Comment (valbers) [reply src/FSharp.Data.GraphQL.Shared/Websockets/Exceptions.fs]: Done.

File: src/FSharp.Data.GraphQL.Shared/Serialization/JsonConverters.fs (hunk 1 above)
Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Shared/Serialization/JsonConverters.fs]: Ah, yes. However the slash disappeared
```suggestion
        String.Join ("\n", graphQLErrorMsgs |> Seq.map (fun err -> err.Message))
```

**Review by xperiandri — CHANGES_REQUESTED** [review]

File: src/FSharp.Data.GraphQL.Shared/Output.fs (hunk 3 above)
Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Shared/Output.fs]: I think that is fine. As `NameValueLookup` is an `Output`

Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Shared/Output.fs]: Or we can go opposide way and move this interface there

Comment (valbers) [reply src/FSharp.Data.GraphQL.Shared/Output.fs]: > As NameValueLookup is an Output

ok. I moved `NameValueLookup` to `Output.fs`.

File: samples/chat-app/server/Program.fs [hunk 6]
```diff
@@ -29,7 +29,6 @@ let main args =
     if app.Environment.IsDevelopment () then
         app.UseGraphQLAltair "/altair" |> ignore
         app.UseGraphQLGraphiQL "/graphiql" |> ignore
-        app.UseGraphQLPlayground "/playground" |> ignore
```
Comment (xperiandri) [inline samples/chat-app/server/Program.fs:32]: The package is not deprecated, maybe just a method is deprecated?

Comment (valbers) [reply samples/chat-app/server/Program.fs:32]: Now that you pointed that out, I just saw [that the whole package `GraphQL.Server.Ui.Playground` is deprecated](https://github.com/graphql-dotnet/server). Using GraphQLPlayground here was [causing the builds to fail](https://github.com/fsprojects/FSharp.Data.GraphQL/actions/runs/12153973380).

Comment (xperiandri) [reply samples/chat-app/server/Program.fs:32]: Ah, I see. The benefit of the Playground is the ability to get schema in DSL format.
If it is deprecated, then let's remove it

Comment (xperiandri) [reply samples/chat-app/server/Program.fs:32]: Then we also need to remove a line in `Packages.props`

**Review by xperiandri — APPROVED** [review]
Looks good. If no more changes In tended I will merge

**Pull request merged**